angular/core/components/thread_page/context_panel/context_panel.coffee

```diff
@@ -0,0 +1,52 @@
+angular.module('loomioApp').directive 'contextPanel', ->
+  scope: {discussion: '='}
+  restrict: 'E'
+  replace: true
+  templateUrl: 'generated/components/thread_page/context_panel/context_panel.html'
+  controller: ($scope, $rootScope, $window, $timeout, AbilityService, Session, ModalService, ChangeVolumeForm, DiscussionForm, ThreadService, MoveThreadForm, PrintModal, DeleteThreadForm, RevisionHistoryModal) ->
+
+    $scope.showContextMenu = ->
+      AbilityService.canChangeThreadVolume($scope.discussion)
+
+    $scope.canChangeVolume = ->
+      Session.user().isMemberOf($scope.discussion.group())
+
+    $scope.openChangeVolumeForm = ->
+      ModalService.open ChangeVolumeForm, model: => $scope.discussion
+
+    $scope.canEditThread = ->
+      AbilityService.canEditThread($scope.discussion)
+
+    $scope.editThread = ->
+      ModalService.open DiscussionForm, discussion: => $scope.discussion
+
+    $scope.muteThread = ->
+      ThreadService.mute($scope.discussion)
+
+    $scope.unmuteThread = ->
+      ThreadService.unmute($scope.discussion)
+
+    $scope.canMoveThread = ->
+      AbilityService.canMoveThread($scope.discussion)
+
+    $scope.moveThread = ->
+      ModalService.open MoveThreadForm, discussion: => $scope.discussion
+
+    $scope.requestPagePrinted = ->
+      unless discussion.allEventsLoaded()
```

Comment:
Do you mind if I consider this out of scope for this feature and make the change soon?